develop/learning-paths/mobile/articles/03-retrieving-data-mobile-sdk/01-retrieve-and-display-guestbooks.markdown

```diff
@@ -0,0 +1,522 @@
+can retrieve guestbooks from the portal. Now you need to call `initActionBar()` 
+in `onCreate`. Place the call immediately above the `getGuestbooks()` call. Then 
+delete the original `Toolbar` initialization code in `onCreate`; it's no longer 
+needed since you included it in `initActionBar`. The first few lines of the 
+`onCreate` method should now look like this: 
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_guestbooks);
+        
+        initActionBar();
+        getGuestbooks();
+
+        ...
+    }
+
+Now you're ready to write the code that puts the guestbooks in the navigation 
+drawer. First, you need to put a 
+[`ListView`](http://developer.android.com/guide/topics/ui/layout/listview.html) 
+in the drawer's layout. You'll use this `ListView` to display the guestbooks. In 
+`activity_guestbooks.xml`, replace the existing 
+[`NavigationView`](https://developer.android.com/reference/android/support/design/widget/NavigationView.html) 
+with this one: 
+
+    <android.support.design.widget.NavigationView
+        android:id="@+id/nav_view"
+        android:layout_width="wrap_content"
+        android:layout_height="match_parent"
+        android:layout_gravity="start"
+        android:fitsSystemWindows="true"
+        app:headerLayout="@layout/nav_header_guestbooks">
+
+        <ListView
+            android:id="@+id/drawer_list"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:choiceMode="singleChoice" />
+
+    </android.support.design.widget.NavigationView>
+
+When compared to the `NavigationView` it replaced, this new `NavigationView` 
+contains the `ListView`, and lacks the `app:menu` attribute. This attribute 
+points to the menu resource file used to create the drawer's items. Since you'll 
+instead use the `ListView` to display the drawer's items (the guestbooks), you 
+don't need to use `app:menu`. 
+
+Now that you've specified the drawer's layout, you need to write the code that 
+controls the drawer. You'll do this in `GuestbooksActivity`. First, add the 
+following variables to this class:
+
+    private DrawerLayout drawer;
+    private ListView drawerListView;
+    private ArrayAdapter _adapter;
+
+As with the Action Bar, you'll initialize the drawer in its own method. Add 
+this method now:
+
+    private void initDrawer() {
+        // drawer initialization
+        drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
+                this, drawer, toolbar, R.string.navigation_drawer_open, 
+                R.string.navigation_drawer_close);
+        drawer.setDrawerListener(toggle);
+        toggle.syncState();
+        
+        // ListView initialization
+        drawerListView = (ListView) findViewById(R.id.drawer_list_view);
```

Comment:
As @ngaskill pointed out, the variable `R.id.drawer_list_view` should instead be `R.id.drawer_list`